controllers/limitador_controller.go

```diff
@@ -109,24 +113,111 @@ func (r *LimitadorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	}
+
+	logger.Info("status updated", "Name", limitadorObj.Name)
+	return
+}
+
+func (r *LimitadorReconciler) checkLimitadorInstanceIsRunning(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) bool {
+	logger := logr.FromContext(ctx)
+	limitadorInstance := &appsv1.Deployment{}
+	limitadorInstanceNamespacedName := client.ObjectKey{ // Its deployment is built after the same name and namespace
+		Namespace: limitadorObj.Namespace,
+		Name:      limitadorObj.Name,
 	}
-	return nil
+	if err := r.Client().Get(ctx, limitadorInstanceNamespacedName, limitadorInstance); err != nil {
+		logger.Error(err, "Failed to get Limitador Instance.")
+		return false
+	}
+
+	return limitadorInstance.Status.ReadyReplicas >= 1
+}
+
+func updateStatusService(limitadorObj *limitadorv1alpha1.Limitador) (changed bool) {
+	changed = false
+	serviceHost := buildServiceHost(limitadorObj)
+	if serviceHost != limitadorObj.Status.Service.Host {
+		limitadorObj.Status.Service.Host = serviceHost
+		limitadorObj.Status.Service.Ports.GRPC = limitadorObj.GRPCPort()
+		limitadorObj.Status.Service.Ports.HTTP = limitadorObj.HTTPPort()
+		changed = true
+	}
+	return
+}
+
+func updateStatusReady(limitadorObj *limitadorv1alpha1.Limitador, ready bool) (changed bool) {
+	status := metav1.ConditionFalse
+	reason := limitadorv1alpha1.StatusReasonServiceNotRunning
+	message := "There's no Limitador Pod running"
+
+	if ready {
+		status = metav1.ConditionTrue
+		reason = limitadorv1alpha1.StatusReasonInstanceRunning
+		message = ""
+	}
+	limitadorObj.Status.Conditions, changed = updateStatusConditions(limitadorObj.Status.Conditions, metav1.Condition{
+		Type:    limitadorv1alpha1.StatusConditionReady,
+		Status:  status,
+		Reason:  reason,
+		Message: message,
+	})
+
+	return
+}
+
+func updateStatusConditions(currentConditions []metav1.Condition, newCondition metav1.Condition) ([]metav1.Condition, bool) {
+	newCondition.LastTransitionTime = metav1.Now()
+
+	if currentConditions == nil {
+		return []metav1.Condition{newCondition}, true
+	}
+
+	for i, condition := range currentConditions {
+		if condition.Type == newCondition.Type {
+			if condition.Status == newCondition.Status {
+				if condition.Reason == newCondition.Reason && condition.Message == newCondition.Message {
+					return currentConditions, false
+				}
+
+				newCondition.LastTransitionTime = condition.LastTransitionTime
+			}
+
+			res := make([]metav1.Condition, len(currentConditions))
+			copy(res, currentConditions)
+			res[i] = newCondition
+			return res, true
+		}
+	}
+
+	return append(currentConditions, newCondition), true
 }
 
-func buildServiceUrl(limitadorObj *limitadorv1alpha1.Limitador) string {
-	return "http://" +
-		limitadorObj.Name + "." +
-		limitadorObj.Namespace + ".svc.cluster.local:" +
-		strconv.Itoa(int(limitadorObj.HTTPPort()))
+func buildServiceHost(limitadorObj *limitadorv1alpha1.Limitador) string {
+	return fmt.Sprintf("%s.%s.svc.cluster.local", limitadorObj.Name, limitadorObj.Namespace)
```

Comment:
It is not the object name, but the service name

Reply:
As the time being, the service is created out of the Limitador CR Name: https://github.com/Kuadrant/limitador-operator/blob/main/pkg/limitador/k8s_objects.go#L34

Maybe we could make it configurable in a followup PR? or good enough for now?

Reply:
Good for now.

the service name should be in a function where it can be reused. Otherwise, if the service name format changes, this code for the status will break

For example:
```go
func LimitadorServiceName(limitadorObj *limitadorv1alpha1.Limitador) string {
       return fmt.Sprintf("limitador-%s", limitadorObj.Name)
}
```

Reply:
Makes sense, will add it to the poll of changes after this one